editor/src/components/canvas/controls/grid-controls-for-strategies.tsx

```diff
@@ -68,136 +88,170 @@ export type GridMeasurementHelperData = {
+      1 / scale,
+      getRoundingFn('nearest-half'),
+    )
+    const frame = Utils.offsetRect(elementRect, Utils.negate(parentRect))
+
+    return {
+      frame: frame,
+      gridTemplateColumns: containerGridProperties.gridTemplateColumns,
+      gridTemplateRows: containerGridProperties.gridTemplateRows,
+      gridTemplateColumnsFromProps: containerGridPropertiesFromProps.gridTemplateColumns,
+      gridTemplateRowsFromProps: containerGridPropertiesFromProps.gridTemplateRows,
+      gap: gap,
+      rowGap: rowGap,
+      columnGap: columnGap,
+      justifyContent: justifyContent,
+      alignContent: alignContent,
+      padding: padding,
+      rows: rows,
+      columns: columns,
+      cells: columns * rows,
+      border: border,
+    }
+  }
+}
+
 export function useGridMeasurentHelperData(
   elementPath: ElementPath,
-): GridMeasurementHelperData | null {
-  return useEditorState(
-    Substores.metadata,
-    (store) => {
-      const element = MetadataUtils.findElementByElementPath(store.editor.jsxMetadata, elementPath)
-
-      const targetGridContainer = MetadataUtils.isGridLayoutedContainer(element) ? element : null
-
-      if (
-        targetGridContainer == null ||
-        targetGridContainer.globalFrame == null ||
-        !isFiniteRectangle(targetGridContainer.globalFrame)
-      ) {
-        return null
-      }
-
-      const columns = getCellsCount(
-        targetGridContainer.specialSizeMeasurements.containerGridProperties.gridTemplateColumns,
-      )
-      const rows = getCellsCount(
-        targetGridContainer.specialSizeMeasurements.containerGridProperties.gridTemplateRows,
-      )
-
-      return {
-        frame: targetGridContainer.globalFrame,
-        gridTemplateColumns:
-          targetGridContainer.specialSizeMeasurements.containerGridProperties.gridTemplateColumns,
-        gridTemplateRows:
-          targetGridContainer.specialSizeMeasurements.containerGridProperties.gridTemplateRows,
-        gap: targetGridContainer.specialSizeMeasurements.gap,
-        rowGap: targetGridContainer.specialSizeMeasurements.rowGap,
-        columnGap: targetGridContainer.specialSizeMeasurements.columnGap,
-        justifyContent: targetGridContainer.specialSizeMeasurements.justifyContent,
-        alignContent: targetGridContainer.specialSizeMeasurements.alignContent,
-        padding: targetGridContainer.specialSizeMeasurements.padding,
-        columns: columns,
-        cells: rows * columns,
-        border: targetGridContainer.specialSizeMeasurements.borderWidths,
-      }
-    },
-    'useGridMeasurentHelperData',
-    fastDeepEqual, //TODO: this should not be needed, but it seems EditorStateKeepDeepEquality is not running, and metadata reference is always updated
+): GridMeasurementHelperData | undefined {
+  const scale = useEditorState(
+    Substores.canvas,
+    (store) => store.editor.canvas.scale,
+    'useGridMeasurentHelperData scale',
   )
+
+  useMonitorChangesToElements([elementPath])
```

Comment:
I'm afraid this is going to rerender the helpers/controls even during interactions when the grid is not in elements-to-rerender

Reply:
If there's a change to the grid, then it should be in elements-to-rerender surely?